flang/include/flang/Lower/PFTBuilder.h

```diff
@@ -622,6 +622,10 @@ struct FunctionLikeUnit : public ProgramUnit {
     return *symbol;
   }
 
+  bool hasSubprogramSymbol() const {
+    return entryPointList[activeEntry].first != nullptr;
```

Comment:
nit: A long time ago, the group decided that comparisons to `nullptr` weren't necessary in flang.

Reply:
I think this can be eliminated and you can use `!funit.isMainProgram()` since all subprograms have an entry label/symbol.

Reply:
Thanks!  Will do.